src/pool.rs

```diff
@@ -230,22 +230,27 @@ impl<T> Pool<T> {
     /// Restores all the garbage elements back to real elements. Effectively
     /// this is a rollback of all the remove operations that happened
     pub fn restore_garbage(&mut self) {
```

Comment:
Need tests.
`Pool::restore_garbage` is only used in `Tree::restore_garbage`, which is not used anywhere else in the library and not covered by tests.